packages/dependency-extraction-webpack-plugin/index.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * WordPress dependencies
+ */
+const { camelCaseDash } = require( '@wordpress/scripts/utils' );
+
+/**
+ * External dependencies
+ */
+const { createHash } = require( 'crypto' );
+const { ExternalsPlugin } = require( 'webpack' );
+const { RawSource } = require( 'webpack-sources' );
+
+const WORDPRESS_NAMESPACE = '@wordpress/';
+
+class DependencyExtractionWebpackPlugin {
+	constructor( options ) {
+		this.options = Object.assign(
+			{
+				injectPolyfill: false,
+				useDefaults: true,
+			},
+			options
+		);
+		this.externalizedDeps = new Set();
+		this.externalsPlugin = new ExternalsPlugin( 'global', [ this.externalizeWpDeps.bind( this ) ] );
+	}
+
+	externalizeWpDeps( context, request, callback ) {
+		let externRootRequest;
+
+		// Handle via options.requestToExternal first
+		if ( 'function' === typeof this.options.requestToExternal ) {
```

Comment:
yoda conditional, isn't this against js coding standards?  Should this be:

```suggestion
		if ( typeof this.options.requestToExternal === 'function' ) {
```